File: lab2(div)/datetime.cpp
#include "datetime.h"
#include <string>
using namespace std;

bool DateTime::isEqual(const DateTime& other) const {
        return year == other.year && month == other.month && day == other.day &&
               hour == other.hour && minute == other.minute;
    }
    
string formatDate(const DateTime& date) {
    return to_string(date.day) + "." + to_string(date.month) + "." + to_string(date.year) + " " + to_string(date.hour) + ":" + to_string(date.minute);
}

string getDayOfWeek(int day) {
    switch(day) {
        case 1:
            return "Monday";
        case 2:
            return "Tuesday";
        case 3:
            return "Wednesday";
        case 4:
            return "Thursday";
        case 5:
            return "Friday";
        case 6:
            return "Saturday";
        case 7:
            return "Sunday";
        default:
            return "Unknown";
    }
}

File: Models/DateTime.cs
namespace Project.Models;

public struct Data
{
    public Data(int day, int month, int year)
    {
        Day = day;
        Month = month;
        Year = year;
    }
    public int Day { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }
    public int Hour { get; set; }
    public int Minute { get; set; }

    public bool IsEqual(Data other)
    {
        return Year == other.Year && Month == other.Month && Day == other.Day &&
               Hour == other.Hour && Minute == other.Minute;
    }
    
    public string FormatDate()
    {
        return $"{Day}.{Month}.{Year} {Hour}:{Minute}";
    }
    
    public string GetDayOfWeek(int day)
    {
        switch (day)
        {
            case 1:
                return "Monday";
            case 2:
                return "Tuesday";
            case 3:
                return "Wednesday";
            case 4:
                return "Thursday";
            case 5:
                return "Friday";
            case 6:
                return "Saturday";
            case 7:
                return "Sunday";
            default:
                return "Unknown";
        }
    }
    
    public static bool operator ==(Data left, Data right)
    {
        return left.IsEqual(right);
    }

    public static bool operator !=(Data left, Data right)
    {
        return !(left == right);
    }

    public override bool Equals(object obj)
    {
        if (obj is Data other)
        {
            return IsEqual(other);
        }
        return false;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Day, Month, Year, Hour, Minute);
    }
    
    
}
